kit/src/layout/chatbar/mod.rs

```diff
@@ -141,11 +148,44 @@ pub fn Chatbar<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
                     value: if cx.props.is_disabled { get_local_text("messages.not-friends")} else { cx.props.value.clone().unwrap_or_default()},
                     onchange: move |(v, _)| cx.props.onchange.call(v),
                     onreturn: move |(v, is_valid, _)| {
+                        if let Some(i) = selected_emoji.write_silent().take() {
+                            if let Some(e) = cx.props.on_emoji_click.as_ref() {
+                                if let Some(p) = cursor_position.read().as_ref() {
+                                    let (emoji, alias) = cx.props.emoji_suggestions[i].clone();
+                                    e.call((emoji, alias,*p));
+                                    return;
+                                }
+                            }
+                        }
                         if is_valid {
                             cx.props.onreturn.call(v);
                         }
                     },
+                    oncursor_update: move |(v,p)| {
+                        if let Some(e) = cx.props.oncursor_update.as_ref() {
+                            e.call((v,p))
+                        }
+                        *cursor_position.write_silent() = Some(p)
+                    },
                     is_disabled: cx.props.is_disabled,
+                    prevent_up_down_arrows: !cx.props.emoji_suggestions.is_empty(),
+                    onup_down_arrow: 
+                        move |code| {
+                            let current = &mut *selected_emoji.write_silent();
```

Comment:
you should obtain the `Some` value from `selected_emoji` here, and if `selected_emoji` is `None`, then `return`. basically stop using `unwrap` unless absolutely necessary.